### PR TITLE
[GH-#83] Added fee to transactions

### LINKED ISSUE #83: add fee to transaction
- [x] add fee value to the transaction
- [x] the fee will should be reduced from `from_acc` e.g. balance 100 --> `from_acc` 99 `fee` 1 
- [x] the miner should be able to collect all fees from transactions in the block in addition to the coinbase

### PR DESCRIPTION
**Pull request by cheezus1**
closes #83

### CODE REVIEW

File: apps/aecore/lib/aecore/chain/chain_state.ex [hunk 1]
```diff
@@ -79,6 +82,25 @@ defmodule Aecore.Chain.ChainState do
       Enum.all?()
   end
 
+  @spec add_fees_to_block_state(list(), map()) :: map()
+  def add_fees_to_block_state(txs, block_state) do
+    pubkey = elem(Keys.pubkey(), 1)
```
Comment (thepiwo) [inline apps/aecore/lib/aecore/chain/chain_state.ex]: Why is the pubkey accessed here?

File: apps/aecore/lib/aecore/chain/chain_state.ex [hunk 2]
```diff
@@ -79,6 +82,25 @@ defmodule Aecore.Chain.ChainState do
       Enum.all?()
   end
 
+  @spec add_fees_to_block_state(list(), map()) :: map()
+  def add_fees_to_block_state(txs, block_state) do
```
Comment (thepiwo) [inline apps/aecore/lib/aecore/chain/chain_state.ex]: What does this function do?

Comment (cheezus1) [reply apps/aecore/lib/aecore/chain/chain_state.ex]: it adds the sum of the fees to the given account in the block state

**Comment by thepiwo** [comment]
I think the `sum_coinbase_transactions` is wrong, it should add only coinbase transactions values and compare with the `Miner.coinbase_transaction_value()` + sum of tx fees

**Pull request merged**